nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/PutKudu.java

```diff
@@ -0,0 +1,130 @@
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(KUDU_MASTERS);
+        properties.add(TABLE_NAME);
+        properties.add(BATCH_SIZE);
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_FAILURE);
+        return rels;
+    }
+
+    @Override
+    protected OperationResponse createPut(ProcessSession session, ProcessContext context, FlowFile flowFile) {
+        try {
+            Set<Map.Entry<String, String>> attrSet = flowFile.getAttributes().entrySet();
+            org.apache.kudu.client.Insert insert = kuduTable.newInsert();
+            Schema colSchema = kuduTable.getSchema();
+            PartialRow row = insert.getRow();
+
+            for (Map.Entry<String, String> entry : attrSet) {
+                String colName = entry.getKey();
+                int colIdx = this.getColumnIndex(colSchema, colName);
+                //getLogger().info("CreatePut: " + colName + " " + entry.getValue() + " - Found Col: " + colIdx);
+                if (colIdx != -1) {
+                    Type colType = colSchema.getColumn(colName).getType();
+                    String val = entry.getValue();
+                    switch (colType.getDataType()) {
+                        case BOOL:
+                            row.addBoolean(colIdx, Boolean.parseBoolean(val));
+                            break;
+                        case FLOAT:
+                            row.addFloat(colIdx, Float.parseFloat(val));
+                            break;
+                        case DOUBLE:
+                            row.addDouble(colIdx, Double.parseDouble(val));
+                            break;
+                        case BINARY:
+                            row.addBinary(colIdx, val.getBytes());
+                            break;
+                        case INT8:
+                        case INT16:
+                            row.addShort(colIdx, Short.parseShort(val));
+                            break;
+                        case INT32:
+                            row.addInt(colIdx, Integer.parseInt(val));
+                            break;
+                        case INT64:
+                            row.addLong(colIdx, Long.parseLong(val));
+                            break;
+                        case STRING:
+                            row.addString(colIdx, val);
+                            break;
+                        default:
+                            throw new IllegalStateException(String.format("unknown column type %s", colType));
+                    }
+                }
+            }
+
+            return kuduSession.apply(insert);
+        } catch (Exception ex) {
+            getLogger().error("CamMach My Error: " + ex.getMessage() + "::" + ex.getStackTrace());
```

Comment:
please alter this line to remove CamMach.   Also use the error(String,Throwable) rather than forcing the stacktrace there.   getLogger().error("Exception occurred while interacting with Kudu due to " + ex.getMessage(), ex);    Is there a more narrow scoped exception or set of exceptions to monitor for rather than Exception itself?